src/lib/notifier.c

```diff
@@ -109,6 +109,9 @@ void notifier_unregister(void *receiver, void *caller, enum notify_id type)
 		    (!caller || handle->caller == caller)) {
 			if (!--handle->num_registrations) {
 				list_item_del(&handle->list);
+#ifdef CONFIG_PROBE
```

Comment:
oh, can you confirm we have the opposite list_add() call for this notifier using the same #ifdef CONFIG_PROBE kconfig ?

Reply:
only for list_item_del(&notify->list[NOTIFIER_ID_BUFFER_PRODUCE]) is used as it is bug for probe buffer to avoid issues for regulator use cases

Reply:
There is probably a real bug here, but this does not look a correct fix. The probe code does notify unregister for NOTIFIER_ID_BUFFER_PRODUCE and NOTIFIER_ID_BUFFER_FREE, so the unconditional list_item_del seems out of place here.

Reply:
Quick look at the API means we need to call ```notifier_unregister()``` and pass in ```NOTIFIER_ID_BUFFER_PRODUCE``` as type.

Reply:
Updated the patch with proper fix and removed the macro config_probe.